philander/potentiometer.py

```diff
@@ -0,0 +1,119 @@
+        :return: An error code indicating either success or the reason of failure.
+        :rtype: ErrorCode
+        """
+        return NotImplementedError
+        
+    def close(self, paramDict):
+        """Close this instance and release hardware resources.
+        
+        Also see: :meth:`.Module.close`.
+        
+        :return: An error code indicating either success or the reason of failure.
+        :rtype: ErrorCode
+        """
+        return NotImplementedError
+    
+    def _digitalize_resistance_value(value, isAbsolute=False, max_resistance=None, isDigital=False, resolution=None):
+        """Converts an either digital, absolute or relative resistance value into a digital value. It also checks for it's validity.
+
+        Also see: :meth:`.potentiometer._eval_resistance_value`.
+        
+        :param int value: Resistance value, interpreted as percentage (0 to 100) by default.\
+        Will be inpreted as absolute resistance in ohms, if isAbsolute is set true.
+        :param bool isAbsolute: Set true if value is in ohms.
+        :return: An error code indicating either success or the reason of failure.
+        :rtype: int
+        """
+        Potentiometer._eval_resistance_value(value, isAbsolute, max_resistance, isDigital, resolution)
+        pot_steps = 2 ** resolution # number of available steps, dependent on number of bits for resolution
+        if isAbsolute:
+            return round((value / max_resistance) * (pot_steps - 1))
+        elif isDigital:
+            return value
+        else:
+            return round((value / 100) * (pot_steps - 1))
+    
+    def _eval_resistance_value(value, isAbsolute=False, max_resistance=None, isDigital=False, resolution=None):
+        """Evaluate values given to set method. Raises error if values are invalid (e.g. over 100% or over maximum resistance).
+        
+        :param int value: Resistance value, interpreted as percentage (0 to 100) by default.\
+        Will be inpreted as absolute resistance in ohms, if isAbsolute is set true.
+        :param bool isAbsolute: Set true if value is in ohms.
+        :return: An error code indicating either success or the reason of failure.
+        :rtype: None
+        """
+        if isAbsolute and (value < 0 or value > max_resistance):
+            print(f"Given absolute resistance value was {value} ohm must be between 0 ohm and the given maximum ({max_resistance} is currently set)")
+            raise ValueError
+        elif isDigital and (value < 0 or value >= (2**resolution)):
+            print(f"Digital value must be between 0 and the given resolution (currently {resolution} bit -> {2**resolution}).")
+            raise ValueError
+        elif not isAbsolute and not isDigital and (value < 0 or value > 100):
+            print("Percentage value must be between 0 and 100.")
+            raise ValueError
+        return None
+    
+    def set(self, value, isAbsolute=False):

```

Comment:
Please check for either:
- one function with two named parameters
- two functions with the same name and different parameters